### DISCUSSION TITLE
Plug in ESP32 modules w/bluetooth LE

### DISCUSSION THREAD

**Discussion by EmilyNerdGirl** [ESPresense/ESPresense#38] [General]
While I don't mind using ESP32 dev boards with a 3D printed enclosure and a plug in power supply, would nice to have a self contained plug in unit running ESPresense. I was looking for an ESP32 based wall module. Looks like some may be coming out soon:
https://www.mokosmart.com/esp32-ble-gateway-mk110/

I couldn't find them on Alexpress yet, but they are on Alibaba.  Not sure if others have found anything similar yet?

**Comment by DTTerastar** [comment]
Very cool! I want one!  We could buy a bunch and pre-flash ESPresence on them and make a store...

**Reply by EmilyNerdGirl** [reply]
I had wondered the same thing. I thought I read they claim to do OEM services so may even be able to preflash a bunch of units....

**Reply by ebroedig** [reply]
I know it's still not an all in one unit but I made a small case so that you can plug a bunch of these around your house and keep it clean still without any cables.  I posted it up at [Printables](https://www.printables.com/model/335469-usb-outlet-esp32-d1-mini-case-for-espresense-and-m).

**Reply by DTTerastar** [reply]
Super cool, i added it to https://github.com/ESPresense/ESPresense.github.io/blob/master/enclosures.md
